### ISSUE TITLE
How to  use Regression by catboost.caret  

### ISSUE THREAD

**Issue by qia99** [catboost/catboost#2239]
I want to use regression analysis  by catboost.caret   in R language. But   I got the   results of classification.
This is  my  code.
library(caret)
library(catboost)

data_train<-as.data.frame(matrix(NA,nrow = 100,ncol = 3))
set.seed(2018)
haha  <-  runif(100)
colnames(data_train)<-c('LABEL','x1','x2')
catboost.caret.Regression

data_train$LABEL<-factor(haha)
data_train$x1<-rnorm(100,0,0.5)
data_train$x2<-rnorm(100,0,0.1)
ctrl=trainControl(method = 'cv',
                  number = 3 )
           

x = data_train[, -1]
Model_catboost<-train(y = data_train$LABEL, x = data_train[, -1],
                      method = catboost.caret,trControl=ctrl)
print(Model_catboost)

This is  the result:
Catboost 

100 samples
  2 predictor
100 classes: '0.00286978343501687', '0.0088143574539572', '0.0338302177842706', '0.0384753795806319', '0.0460559336934239', '0.0521734403446317', '0.0605853858869523', '0.0646366507280618', '0.0647393725812435', '0.0691020083613694', '0.093992194859311', '0.0949368453584611', '0.0968799674883485', '0.0970170998480171', '0.111288217129186', '0.114261748734862', '0.130012104287744', '0.141128375194967', '0.14136787969619', '0.144876915263012', '0.151498265331611', '0.154708412941545', '0.175132384989411', '0.192185702733696', '0.197433612542227', '0.209981348598376', '0.218055207515135', '0.228056290419772', '0.235274362144992', '0.260100793326274', '0.26375067862682', '0.268232818692923', '0.270736463367939', '0.275984284700826', '0.301048603840172', '0.307888902723789', '0.312140962807462', '0.318238113308325', '0.327880936674774', '0.336153471143916', '0.350549737457186', '0.359632111852989', '0.393212184775621', '0.395014614565298', '0.395615968387574', '0.398660670500249', '0.405117745744064', '0.409663012251258', '0.418135951273143', '0.440919016487896', '0.463723270455375', '0.474314193474129', '0.485334310447797', '0.489468237617984', '0.49020482157357', '0.50618012319319', '0.535928548080847', '0.540995186194777', '0.546849492704496', '0.55290412530303', '0.568360368022695', '0.581005929736421', '0.598311740672216', '0.598782849730924', '0.599246623693034', '0.601871956605464', '0.604347600601614', '0.606758858077228', '0.615007577696815', '0.634179883170873', '0.645794287556782', '0.662541398778558', '0.664538606768474', '0.678215388208628', '0.701299199601635', '0.737955683842301', '0.739065114408731', '0.760730505455285', '0.76191659620963', '0.762034389888868', '0.763491135556251', '0.772971612401307', '0.774272713810205', '0.806027805432677', '0.828400381840765', '0.845679219346493', '0.854141466785222', '0.862599012441933', '0.88073713472113', '0.896860512206331', '0.909029520582408', '0.920807133661583', '0.924356027971953', '0.945775595027953', '0.951702587306499', '0.958654714981094', '0.962912647984922', '0.966548021417111', '0.982112286379561', '0.993275427026674' 

No pre-processing
Resampling: Cross-Validated (3 fold) 
Summary of sample sizes: 59, 68, 73 
Resampling results across tuning parameters:

  depth  learning_rate  Accuracy    Kappa        
  2      0.04978707     0.00000000  -0.0089157193
  2      0.13533528     0.01234568   0.0030667596
  2      0.36787944     0.01234568   0.0021532200
  2      1.00000000     0.01234568   0.0033828548
  4      0.04978707     0.01234568   0.0029810236
  4      0.13533528     0.01234568   0.0031107398
  4      0.36787944     0.02469136   0.0147728295
  4      1.00000000     0.02469136   0.0137158719
  6      0.04978707     0.01234568  -0.0001121247
  6      0.13533528     0.02469136   0.0129049898
  6      0.36787944     0.02469136   0.0136743483
  6      1.00000000     0.02469136   0.0141101904

Tuning parameter 'iterations' was held constant at a value of 100
 1e-06
Tuning parameter 'rsm' was held constant at a value of 0.9

Tuning parameter 'border_count' was held constant at a value of 255
Accuracy was used to select the optimal model using the largest value.
The final values used for the model were depth = 4, learning_rate
 = 0.3678794, iterations = 100, l2_leaf_reg = 1e-06, rsm = 0.9
 and border_count = 255.
So ,  what can I do , if I want to use regression analysis  by catboost.caret   in R language.
